### PR TITLE
add quota failure

### PR DESCRIPTION
**Pull request by EarthmanT**
add delete props on group,eni,vm
#135

### CODE REVIEW

File: cloudify_awssdk/common/decorators.py [hunk 1]
```diff
@@ -135,6 +135,16 @@ def wrapper_inner(**kwargs):
                     ctx.instance.runtime_properties[
                         'resource_config'] = resource_config
                     ctx.instance.runtime_properties[EXT_RES_ID] = resource_id
+                    _, _, _, operation_name = ctx.operation.name.split('.')
+                    if operation_name not in ['delete', 'create'] and \
```
Comment (mabuaisha) [inline cloudify_awssdk/common/decorators.py]: @EarthmanT Why we need just to ignore `delete` and `create`

File: cloudify_awssdk/common/decorators.py [hunk 2]
```diff
@@ -135,6 +135,16 @@ def wrapper_inner(**kwargs):
...
+                            ctx.node.properties['verify_external_resource'] \
+                            and not \
+                            kwargs['iface'].resource_id_in_properties():
+                        raise NonRecoverableError(
+                            'Resource type {0} resource_id '
+                            '{1} not found.'.format(
+                                kwargs['resource_type'],
+                                kwargs['iface'].resource_id))
```
Comment (mabuaisha) [inline cloudify_awssdk/common/decorators.py]: @EarthmanT Is this a special case for IP or we can apply it on ant other AWS resources ?

File: plugin.yaml [hunk 3]
```diff
@@ -597,6 +597,12 @@ dsl_definitions:
       type: boolean
       default: false
 
+  verify_external_resource: &verify_external_resource
```
Comment (mabuaisha) [inline plugin.yaml]: Cannot we depend on `use_external_resource` do we need to add this ?

File: cloudify_awssdk/common/decorators.py (hunk 1 above)
Comment (EarthmanT) [reply cloudify_awssdk/common/decorators.py]: For all resources, even existing resources, we do not expect this property to exist during `create` or during `delete`. But already in `configure`, `start`, and `stop`, we do expect it to exist.
I'm not crazy about this method - but I'm not sure of another way to make sure we do not run this function in those operations.

File: cloudify_awssdk/common/decorators.py (hunk 2 above)
Comment (EarthmanT) [reply cloudify_awssdk/common/decorators.py]: Yes I want to. I just did not want to do all of the work and testing, without first checking out the solution.

File: plugin.yaml (hunk 3 above)
Comment (EarthmanT) [reply plugin.yaml]: The problem here is I am not sure right now that this method can be used with every resource type.

**Pull request merged**